pagefaultgames/pokerogue
# [Bug] Fixing crash that can occur when loading variant sprite,

## What are the changes the user will see?
- Crash will no longer happen

## Why am I making these changes?
- Game can crash when failing to load variant sprites

## What are the changes from a developer perspective?
- Upon and error, attempt to fall back to non-experimental sprite. If that fails as well the additional error handling will prevent the game from crashing.
- similar issue to https://github.com/pagefaultgames/pokerogue/pull/4266

## How to test the changes?
- Turn on experimental sprites.
- With the following save, after evolving can observe the error. With loading from website response error handling will work. However, if an error is thrown without a response then crash happens (ie - happens in the offline client more readily)
[sessionData_Guest-evolve-crash.prsv.zip](https://github.com/user-attachments/files/17046252/sessionData_Guest-evolve-crash.prsv.zip)

## Bug reports
- https://github.com/pagefaultgames/pokerogue/issues/4316
- Discord issue that covers multiple similar crashes: https://discord.com/channels/1125469663833370665/1282438230578430073

## Checklist
